Add Kibana upgrade notes page (#193268)
elastic/kibana: docs/upgrade-notes.asciidoc

[[breaking-changes-summary]]
== Upgrade notes

////
USE THE FOLLOWING TEMPLATE to add entries to this document.
1. Copy from "[discrete]" to the last "====" included.
2. Paste and edit the template in the right area section. Most recent entries should be at the top of the section. 
3. Make sure to edit the anchor ID [[REPO-PR]] with proper values, then add the corresponding entry to the doc link service https://github.com/elastic/kibana/blob/main/packages/kbn-doc-links/src/get_doc_links.ts:

> codeId: `${KIBANA_DOCS}breaking-changes-summary.html#REPO-PR` // replace `codeId` with the ID of your choice that you want to call in the Upgrade Assistant code, and replace `REPO-PR` with the corresponding values you edited in this document.

Reviewer: Not 100% I follow what `codeId` is

Author: That's the ID assigned to the link in the doc link service file, that can be called in the code instead of hardcoding the link. I slightly edited the template and added a link to a small doc explaining the docs link service, hopefully this will be clearer.

> [discrete]

Reviewer: Nitnit + opinion: the flow a bit better if we first see the template

Author: ✅

[[REPO-PR]]
.[FEATURE] TITLE TO DESCRIBE THE CHANGE. (VERSION)
[%collapsible]
====
*Details* +
ADD MORE DETAILS ON WHAT IS CHANGING AND A LINK TO THE PR INTRODUCING THE CHANGE

*Impact* +
ADD INFORMATION ABOUT WHAT THIS CHANGE WILL BREAK FOR USERS

*Action* +
ADD INSTRUCTIONS FOR USERS LOOKING TO UPGRADE. HOW CAN THEY WORK AROUND THIS?
====

> ////

Author: @lcawl @jloleysens @rudolf Here is the file where we can compile breaking changes and deprecation announcements at the same place. Based on our discussion yesterday, can you let me know your thoughts on these 2 things:
- The content template just here at the top of the file. We could ask teams to update this file following that template when adding deprecations or breaking changes. In that template are instructions on what to do to populate links in the Upgrade Assistant through the doc links service file.
- The IDs for each item. Right now they are not entirely consistent with each other because they were compiled from existing sources, but for each new addition to this file, we can use a fixed pattern, such as `[[kibana-PR_number]]` as proposed in the template.

Reviewer 2: That all seems like a good step forward as we ultimately get move toward more consistency and automation across our release docs.

Reviewer: This looks like a great start!

> This page summarizes all breaking changes and deprecation notices introduced to {kib} version 8 since 8.0. Review these changes before upgrading Kibana to mitigate the impact to your application.

Reviewer 3: ```suggestion
Before you upgrade, review the breaking changes and deprecations introduced in {kib} 8.x, then mitigate the impact. 
```

> NOTE: For the complete Elastic Security solution release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Reviewer 3: ```suggestion
For Elastic Security release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
```

[float]
=== Breaking changes

[float]
==== Kibana APIs

[discrete]
[[breaking-162506]]
.Get case metrics APIs became internal. (8.10)
[%collapsible]
====
*Details* +
The get case metrics APIs are now internal. For more information, refer to ({kibana-pull}162506[#162506]).
====

[discrete]
[[breaking-155470]]
.Removed legacy project monitor API. (8.8)
[%collapsible]
====
*Details* +
The project monitor API for Synthetics in Elastic Observability has been removed. For more information, refer to {kibana-pull}155470[#155470].

*Impact* +
In 8.8.0 and later, an error appears when you use the project monitor API.
====

[discrete]
[[breaking-147616]]
.Removed the `current_upgrades` endpoint. (8.7)
[%collapsible]
====
*Details* +

> The `api/fleet/current_upgrades` endpoint has been removed. For more information, refer to {kibana-pull}147616[#147616].

Reviewer 4: ```suggestion
The `/api/fleet/current_upgrades` endpoint has been removed. For more information, refer to {kibana-pull}147616[#147616].
```

*Impact* +

> When you upgrade to 8.7.0, use the `/action_status` endpoint.

Reviewer 4: I'm not sure that `/action_status` is actually the right path. Is it `/agents/action_status`? Perhaps @kpollich or someone from the Fleet team could confirm.

Author: ```suggestion
When you upgrade to 8.7.0, use the `api/fleet/agents/action_status` endpoint.
```
Looked it up in our API docs https://www.elastic.co/docs/api

*Impact* +
If you do two or more requests in less than 10 seconds, the subsequent requests fail with `429 Too Many Requests`.
Wait 10 seconds before uploading again.
This change could potentially break automations for users that rely on frequent package uploads.
====

[discrete]
[[breaking-176879]]
.[Fleet]Removed conditional topics for Kafka outputs. (8.13)
[%collapsible]
====
*Details* +
The Kafka output no longer supports conditional topics. For more information, refer to ({kibana-pull}176879[#176879]).
====

[discrete]
[[breaking-176443]]
.[Fleet]Most Fleet installed integrations are now read-only and labelled with a *Managed* tag in the Kibana UI. (8.13)
[%collapsible]
====
*Details* +

Integration content installed by {fleet} is no longer editable. This content is tagged with *Managed* in the {kib} UI, and is Elastic managed. This content cannot be edited or deleted. However, managed visualizations, dashboards, and saved searches can be cloned. The clones can be customized.

When cloning a dashboard the cloned panels become entirely independent copies that are unlinked from the original configurations and dependencies.

> Managed content relating to specific visualization editors such as Lens, TSVB, and Maps, the clones retain the original reference configurations. The same applies to editing any saved searches in a managed visualization.

Reviewer 4: ```suggestion
For managed content relating to specific visualization editors such as Lens, TSVB, and Maps, the clones retain the original reference configurations. The same applies to editing any saved searches in a managed visualization.
```

For more information, refer to ({kibana-pull}172393[#172393]).
====

[discrete]
[[breaking-167085]]
.[Fleet] Improved config output validation for default output. (8.11)
[%collapsible]
====
*Details* +
Improve config output validation to not allow to defining multiple default outputs in {kib} configuration. For more information, refer to ({kibana-pull}167085[#167085]).
====

[discrete]
[[breaking-138677]]
.[Fleet] Removed the `package_policies` field from the agent policy saved object. (8.5)
[%collapsible]
====
*Details* +
The bidirectional foreign key between agent policy and package policy has been removed. For more information, refer to {kibana-pull}138677[#138677].

*Impact* +
The agent policy saved object no longer includes the `package_policies` field.
====

[discrete]
[[breaking-135669]]
.[Fleet] xpack.agents.* are now uneditable in UI when defined in kibana.yml. (8.4)
[%collapsible]
====
*Details* +

> When you configure `setxpack.fleet.agents.fleet_server.hosts` and `xpack.fleet.agents.elasticsearch.hosts` in kibana.yml, you are unable to update the fields on the Fleet UI.

Reviewer 4: I am pretty sure this is just a typo:
```suggestion
When you configure `xpack.fleet.agents.fleet_server.hosts` and `xpack.fleet.agents.elasticsearch.hosts` in kibana.yml, you are unable to update the fields on the Fleet UI.
```

For more information, refer to {kibana-pull}135669[#135669].

*Impact* +

> To configure `setxpack.fleet.agents.fleet_server.hosts` and `xpack.fleet.agents.elasticsearch.hosts` on the Fleet UI, avoid configuring the settings in kibana.yml.

Reviewer 4: ```suggestion
To configure `xpack.fleet.agents.fleet_server.hosts` and `xpack.fleet.agents.elasticsearch.hosts` on the Fleet UI, avoid configuring the settings in kibana.yml.
```

 {
   packagePolicyIds: [...]
 }
--
====

// General settings

[discrete]
[[breaking-111535]]
.[General settings] Removed `CONFIG_PATH` and `DATA_PATH` environment variables. (8.0)
[%collapsible]
====
*Details* +
The `CONFIG_PATH` and `DATA_PATH` environment variables have been removed. For more information, refer to {kibana-pull}111535[#111535].

*Impact* +
Replace the `CONFIG_PATH` environment variable with `KBN_PATH_CONF`, and replace `DATA_PATH` with the `path.data` setting.
====

[discrete]
[[breaking-114379]]
.[General settings] Removed support for csp.rules configuration. (8.0)
[%collapsible]
====
*Details* +
Support for the `csp.rules` configuration property has been removed. For more information, refer to {kibana-pull}114379[#114379].

*Impact* +
Configuring the default `csp.script_src`, `csp.workers_src`, and `csp.style_src` values is not required.
====

[discrete]
[[breaking-113653]]
.[General settings] Changed and removed deprecated core settings and deprecated settings from core plugins. (8.0)
[%collapsible]
====
*Details* +
The deprecation notice for `server.cors` has changed from `level:critical` to `level:warning`.

The following settings have changed:

> * The `xpack.banners.placement` value has been renamed

Reviewer 4: ```suggestion
* The `xpack.banners.placement` value of `header` has been renamed to `top`
```

> * The `newsfeed.defaultLanguage` newsfeed item retrieval method and default language has changed

Reviewer 4: It's confusing having this line because the setting was removed as outlined below on line 440 and explained on line 450. Suggest removing this one

Author: ```suggestion

```

*Impact* +
Before you upgrade to 8.0.0, remove `plugins.scanDirs` from kibana.yml.
====

[discrete]
[[breaking-113296]]
.[General settings] Removed support for `optimize.*` settings. (8.0)
[%collapsible]
====
*Details* +
The legacy `optimize.*` settings have been removed. If your configuration uses the following legacy `optimize.*` settings, {kib} fails to start:

* `optimize.lazy`
* `optimize.lazyPort`
* `optimize.lazyHost`
* `optimize.lazyPrebuild`
* `optimize.lazyProxyTimeout`
* `optimize.enabled`
* `optimize.bundleFilter`
* `optimize.bundleDir`
* `optimize.viewCaching`
* `optimize.watch`
* `optimize.watchPort`
* `optimize.watchHost`
* `optimize.watchPrebuild`
* `optimize.watchProxyTimeout`
* `optimize.useBundleCache`
* `optimize.sourceMaps`
* `optimize.workers`
* `optimize.profile`
* `optimize.validateSyntaxOfNodeModules`

For more information, refer to {kibana-pull}113296[#113296].

*Impact* +
To run the `@kbn/optimizer` separately in development, pass `--no-optimizer` to `yarn start`. For more details, refer to {kibana-pull}73154[#73154].
====

[discrete]
[[breaking-113173]]
.[General settings] Removed `so/server/es` settings. (8.0)
[%collapsible]
====
*Details* +

> Some of the `so/server/es` settings have been removed. If your configuration uses the following settings, {kib} fails to start:

Reviewer 4: ```suggestion
Some of the `savedObjects`, `server`, and `elasticsearch` settings have been removed. If your configuration uses the following settings, {kib} fails to start:
```

[%collapsible]
====
*Details* +
The logging configuration and log output format has changed. For more information, refer to {kibana-pull}112305[#112305].

*Impact* +
Use the new <<logging-configuration,logging system configuration>>.
====

[discrete]
[[breaking-109798]]
.[General settings] Removed `kibana.defaultAppId` setting. (8.0)
[%collapsible]
====
*Details* +
The deprecated `kibana.defaultAppId` setting in kibana.yml, which is also available as `kibana_legacy.defaultAppId`, has been removed. For more information, refer to {kibana-pull}109798[#109798].

*Impact* +
When you upgrade, remove `kibana.defaultAppId` from your kibana.yml file. To configure the default route for users when they enter a space, use the <<defaultroute,`defaultRoute`>> in *Advanced Settings*.
====

[discrete]
[[breaking-109350]]
.[General settings] Removed `courier:batchSearches` setting. (8.0)
[%collapsible]
====
*Details* +
The deprecated `courier:batchSearches` setting in *Advanced Settings* has been removed. For more information, refer to {kibana-pull}109350[#109350].

*Impact* +
When you upgrade, the `courier:batchSearches` setting will no longer be available.
====

[discrete]
[[breaking-108111]]

> .[General settings] Removed `xpack.task_manager.index` setting. (8.0)

Reviewer 4: already mentioned above on L559

> [discrete]
> [[breaking-108111]]
> .[General settings] Removed `xpack.task_manager.index` setting. (8.0)
> [%collapsible]
> ====
> *Details* +
> The deprecated `xpack.task_manager.index` setting in kibana.yml has been removed. For more information, refer to {kibana-pull}108111[#108111].
> 
> *Impact* +
> When you upgrade, remove `xpack.task_manager.index` from your kibana.yml file.
> ====

Author: ```suggestion

```

[discrete]
[[breaking-106061]]
.[General settings] New session timeout defaults. (8.0)
[%collapsible]
====
*Details* +
The default values for the session timeout `xpack.security.session.{lifespan|idleTimeout}` settings have changed. For more information, refer to {kibana-pull}106061[#106061]

*Impact* +

> Use the following default values:

Reviewer 4: ```suggestion
The new default values are as follows:
```

====

// Osquery

[discrete]
[[breaking-134855]]
.[Osquery] "All" base privilege option now also applies to Osquery. (8.3)
[%collapsible]
====
*Details* +
The Osquery {kib} privilege has been updated, so that when the *Privileges for all features level* is set to *All*, this now applies *All* to Osquery privileges as well. Previously, users had to choose the *Customize* option to grant any access to Osquery. For more information, refer to {kibana-pull}130523[#130523].

*Impact* +
This impacts user roles that have *Privileges for all features* set to *All*. After this update, users with this role will have access to the Osquery page in {kib}. However, to use the Osquery feature fully, these requirements remain the same: users also need Read access to the logs-osquery_manager.result* index and the Osquery Manager integration must be deployed to Elastic Agents.
====

// Saved objects

[discrete]
[[breaking-118300]]
.[Saved objects] Fail migrations for saved objects with unknown types. (8.0)
[%collapsible]
====
*Details* +
Unknown saved object types now cause {kib} migrations to fail. For more information, refer to {kibana-issue}107678[#107678].

*Impact* +

> To complete the migration, re enable plugins or delete documents from the index in the previous version.

Reviewer 4: ```suggestion
To complete the migration, re-enable plugins or delete documents from the index in the previous version.
```

====

[discrete]
.A new sub-feature privilege to control user access to the cases settings. (8.12)
[%collapsible]
====
*Details* +
Roles with at least a sub-feature privilege configured will not have access to the cases setting like they had previously. All roles without a sub-feature privilege configured will not be affected. For more information, refer to ({kibana-pull}170635[#170635]).
====

[discrete]
.Improved config output validation for default output. (8.11)
[%collapsible]
====
*Details* +
Improve config output validation to not allow to defining multiple default outputs in {kib} configuration. For more information, refer to ({kibana-pull}167085[#167085]).
====

[discrete]
.Converted filterQuery to KQL.(8.11)
[%collapsible]
====
*Details* +
Converts `filterQuery` to a KQL query string. For more information, refer to ({kibana-pull}161806[#161806]).
====

[discrete]
.Moved doc_root.vulnerability.package to doc_root.package (ECS). (8.11)
[%collapsible]
====
*Details* +
This updates all instances of vulnerability.package to the ECS standard package fieldset. For more information, refer to ({kibana-pull}164651[#164651]).
====

[discrete]

> .Introduced new summary search capabilities that will cause existing SLOs to stop working. (8.10)

Author: ```suggestion
.Introduced new summary search capabilities that will cause SLOs created before 8.10 to stop working. (8.10)
```

[%collapsible]
====
*Details* +

Integration content installed by {fleet} is no longer editable. This content is tagged with *Managed* in the {kib} UI, and is Elastic managed. This content cannot be edited or deleted. However, managed visualizations, dashboards, and saved searches can be cloned. The clones can be customized.

When cloning a dashboard the cloned panels become entirely independent copies that are unlinked from the original configurations and dependencies.

Managed content relating to specific visualization editors such as Lens, TSVB, and Maps, the clones retain the original reference configurations. The same applies to editing any saved searches in a managed visualization.

For more information, refer to ({kibana-pull}172393[#172393]).
====

[discrete]
.Removed `is_nan`, `is_finite`, and `is_infinite` functions from {esql}. (8.13)
[%collapsible]
====
*Details* +
These functions have been removed from {esql} queries as they are not supported. Errors would be thrown when trying to use them. For more information, refer to ({kibana-pull}174674[#174674]).
====

[discrete]
.New SLO architecture. (8.12)
[%collapsible]
====
*Details* +

> We introduce a breaking change in the SLO features that will break any SLOs created before 8.12. These SLOs have to be manually reset through an API until we provide a UI for it. The data aggregated over time (rollup) is still available in the sli v2 index, but won't be used for summary calculation when reset.

Reviewer 2: ```suggestion
We introduced a breaking change in the SLO features that will break any SLOs created before 8.12. These SLOs have to be manually reset through an API until we provide a UI for it. The data aggregated over time (rollup) is still available in the SLI v2 index, but won't be used for summary calculation when reset.
```

[%collapsible]
====
*Details* +
Roles with at least a sub-feature privilege configured will not have access to the cases setting like they had previously. All roles without a sub-feature privilege configured will not be affected. For more information, refer to ({kibana-pull}170635[#170635]).
====

[discrete]
.Improved config output validation for default output. (8.11)
[%collapsible]
====
*Details* +
Improve config output validation to not allow to defining multiple default outputs in {kib} configuration. For more information, refer to ({kibana-pull}167085[#167085]).
====

[discrete]
.Converted filterQuery to KQL.(8.11)
[%collapsible]
====
*Details* +
Converts `filterQuery` to a KQL query string. For more information, refer to ({kibana-pull}161806[#161806]).
====

[discrete]
.Moved doc_root.vulnerability.package to doc_root.package (ECS). (8.11)
[%collapsible]
====
*Details* +
This updates all instances of vulnerability.package to the ECS standard package fieldset. For more information, refer to ({kibana-pull}164651[#164651]).
====

[discrete]
.Introduced new summary search capabilities that will cause existing SLOs to stop working. (8.10)
[%collapsible]
====
*Details* +

> New summary search capabilities introduce breaking changes in various places, and we have decided not to handle backward compatibility:

Reviewer 2: > we have decided not to handle backward compatibility:

Is there a better way we can word this? I know this was pulled from elsewhere but sounds like bad messaging.

Author: ```suggestion

```
It's actually repeating the title so we can remove this line

* SLO find API body parameters have changed.
* The index mapping used by the rollup data has changed, and we have added a summary index that becomes the new source of truth for search.
* The rollup transforms have been updated, but existing SLO with their transforms won't be updated.

If some SLOs have been installed in a prior version at 8.10, they won't work after migrating to 8.10. There are two approaches to handle this breaking change. The recommended route is to delete all SLOs before migrating to 8.10. The alternative is to migrate to 8.10 and manually remove the SLOs.

*Removing SLOs before migrating to 8.10*

> Use the SLO UI or the SLO delete API to delete all existing SLOs. This takes care of the Saved Object, Transform and rollup data. When all SLOs have been deleted, then delete the residual rollup indices: `.slo-observability.sli-v1*`. Note that this is v1.

Reviewer 2: ```suggestion
Use the SLO UI or the SLO delete API to delete all existing SLOs. This takes care of the saved object, transform and rollup data. When all SLOs have been deleted, then delete the residual rollup indices: `.slo-observability.sli-v1*`. Note that this is v1.
```

*Impact* +
Before you upgrade to 8.0.0, remove `console.ssl` from kibana.yml.
====

[discrete]
.Removed support for `monitoring.cluster_alerts.allowedSpaces`. (8.0)
[%collapsible]
====
*Details* +
The `monitoring.cluster_alerts.allowedSpaces` setting, which {kib} uses to create Stack Monitoring alerts, has been removed. For more information, refer to {kibana-pull}123229[#123229].

*Impact* +
Before you upgrade to 8.0.0, remove `monitoring.cluster_alerts.allowedSpaces` from kibana.yml.
====

[float]
=== Deprecation notices

The following functionality is deprecated and will be removed at a future date. Deprecated functionality 
does not have an immediate impact on your application, but we strongly recommend you make the necessary 
updates to avoid use of deprecated features.

Use the **Kibana Upgrade Assistant** to prepare for your upgrade to the next version of the Elastic Stack. 
The assistant identifies deprecated settings in your configuration and guides you through the process of 
resolving issues if any deprecated features are enabled. 
To access the assistant, go to **Stack Management** > **Upgrade Assistant**.

> [discrete]
> .Uptime is deprecated in 8.15.0 and will be removed in a future version. (8.15)
> [%collapsible]
> ====
> *Details* +
> The Uptime app is already hidden from Kibana when there is no recent Heartbeat data, and will be completely removed in early 2026. You should migrate to Synthetics as an alternative. For more details, refer to the {observability-guide}/uptime-intro.html[Uptime documentation].
> ====

Author: ```suggestion
[discrete]
.Uptime is deprecated in 8.15.0. (8.15)
[%collapsible]
====
*Details* +
The Uptime app is already hidden from Kibana when there is no recent Heartbeat data. Migrate to Synthetics as an alternative. For more details, refer to the {observability-guide}/uptime-intro.html[Uptime documentation].
====
```
Removing information about Uptime removal as per Sophie's note in the linked issue.

*Impact* +
The `/agent_status` Fleet API now returns the following statuses:

* `all` &mdash; All active and inactive
* `active` &mdash; All active
====

[discrete]
.Deprecated Elastic Synthetics integration. (8.8)
[%collapsible]
====
*Details* +
The Elastic Synthetics integration is deprecated. For more information, refer to {kibana-pull}149506[#149506].

*Impact* +
To monitor endpoints, pages, and user journeys, go to **{observability}** -> **Synthetics (beta)**.
====

[discrete]
.Deprecated ApiKey authentication for interactive users. (8.4)
[%collapsible]
====
*Details* +
The ability to authenticate interactive users with ApiKey via a web browser has been deprecated, and will be removed in a future version.

For more information, refer to {kibana-pull}136422[#136422].

*Impact* +
To authenticate interactive users via a web browser, use <<kibana-authentication,another authentication method>>. Use API keys only for programmatic access to {kib} and {es}.
====

[discrete]
.Removed `apm_user`. (8.3)
[%collapsible]
====
*Details* +
Removes the `apm_user` role. For more information, check {kibana-pull}132790[#132790].

*Impact* +

> The `apm_user`role is replaced with the `viewer` and `editor` built-in roles.

Reviewer 2: ```suggestion
The `apm_user` role is replaced with the `viewer` and `editor` built-in roles.
```
